domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLogger.kt

```diff
@@ -0,0 +1,167 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import org.oppia.android.app.model.EventLog
+import org.oppia.android.app.model.EventLog.FeatureFlagContext
+import org.oppia.android.app.model.EventLog.FeatureFlagContext.FeatureFlagItem
+import org.oppia.android.util.platformparameter.APP_AND_OS_DEPRECATION
+import org.oppia.android.util.platformparameter.DOWNLOADS_SUPPORT
+import org.oppia.android.util.platformparameter.EDIT_ACCOUNTS_OPTIONS_UI
+import org.oppia.android.util.platformparameter.EXTRA_TOPIC_TABS_UI
+import org.oppia.android.util.platformparameter.EnableAppAndOsDeprecation
+import org.oppia.android.util.platformparameter.EnableDownloadsSupport
+import org.oppia.android.util.platformparameter.EnableEditAccountsOptionsUi
+import org.oppia.android.util.platformparameter.EnableExtraTopicTabsUi
+import org.oppia.android.util.platformparameter.EnableFastLanguageSwitchingInLesson
+import org.oppia.android.util.platformparameter.EnableInteractionConfigChangeStateRetention
+import org.oppia.android.util.platformparameter.EnableLearnerStudyAnalytics
+import org.oppia.android.util.platformparameter.EnableLoggingLearnerStudyIds
+import org.oppia.android.util.platformparameter.EnablePerformanceMetricsCollection
+import org.oppia.android.util.platformparameter.EnableSpotlightUi
+import org.oppia.android.util.platformparameter.FAST_LANGUAGE_SWITCHING_IN_LESSON
+import org.oppia.android.util.platformparameter.INTERACTION_CONFIG_CHANGE_STATE_RETENTION
+import org.oppia.android.util.platformparameter.LEARNER_STUDY_ANALYTICS
+import org.oppia.android.util.platformparameter.LOGGING_LEARNER_STUDY_IDS
+import org.oppia.android.util.platformparameter.PERFORMANCE_METRICS_COLLECTION
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.platformparameter.SPOTLIGHT_UI
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Convenience logger for feature flags.
+ *
+ * This logger is meant to be used to log the current status of all feature flags once after the app
```

Comment:
Nit: clarity
```suggestion
 * This logger is meant to be used for feature flag-related logging  on every app launch. It is primarily used within the ApplicationLifeCycleObserver to log the status of feature flags in a given app session.
```

I believe that any future feature flag related logging will be done via this logger.

Reply:
I have modified the comment to include this.